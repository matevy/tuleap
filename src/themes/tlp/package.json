{
  "author": "Enalean Team",
  "name": "@tuleap/tlp",
  "description": "Tuleap CSS framework",
  "license": "GPL-2.0-or-later",
  "main": "index.js",
  "types": "./src/index.d.ts",
  "private": true,
  "dependencies": {
    "@formatjs/intl-getcanonicallocales": "^1.5.1",
    "@formatjs/intl-numberformat": "^5.7.1",
    "@formatjs/intl-pluralrules": "^3.5.1",
    "@formatjs/intl-relativetimeformat": "^7.3.1",
    "@webcomponents/custom-elements": "^1.4.2",
    "flatpickr": "^4.6.6",
    "jquery": "^3.5.1",
    "popper.js": "^1.16.1",
    "select2": "^4.0.13"
  },
  "devDependencies": {
    "codemirror": "^5.57.0",
    "dompurify": "^2.2.2"
  },
  "config": {
    "bin": "../../../node_modules/.bin"
  },
  "scripts": {
    "test": "$npm_package_config_bin/jest"
  }
}
