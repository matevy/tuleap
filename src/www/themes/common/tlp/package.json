{
  "name": "tlp",
  "version": "0.0.1",
  "description": "Tuleap CSS framework",
  "author": "Enalean",
  "license": "GPL-2.0-or-later",
  "main": "index.js",
  "private": true,
  "dependencies": {
    "core-js": "^3.2.0",
    "dom4": "^2.1.5",
    "flatpickr": "^4.6.2",
    "jquery": "2.1.0",
    "popper.js": "^1.14.7",
    "regenerator-runtime": "^0.13.3",
    "select2": "^4.0.10",
    "whatwg-fetch": "^3.0.0"
  },
  "devDependencies": {
    "@types/select2": "^4.0.48",
    "codemirror": "^5.48.4",
    "dompurify": "^2.0.7"
  },
  "types": "./src/index.d.ts",
  "config": {
    "bin": "../../../../../node_modules/.bin"
  },
  "scripts": {
    "build": "$npm_package_config_bin/webpack --config webpack.prod.js",
    "test": "$npm_package_config_bin/jest",
    "watch": "$npm_package_config_bin/concurrently '$npm_package_config_bin/webpack --config webpack.dev.js --watch' 'CI=true npm test -- --watch'"
  }
}
