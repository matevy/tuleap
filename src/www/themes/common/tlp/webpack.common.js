/*
 * Copyright (c) Enalean, 2017-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

const path = require("path");
const polyfills_for_fetch = require("../../../../../tools/utils/scripts/ie11-polyfill-names.js")
    .polyfills_for_fetch;
const webpack_configurator = require("../../../../../tools/utils/scripts/webpack-configurator.js");

let entry_points = {
    "tlp-en_US": polyfills_for_fetch.concat(["dom4", "./src/index.en_US.js"]),
    "tlp-fr_FR": polyfills_for_fetch.concat(["dom4", "./src/index.fr_FR.js"])
};

const colors = ["orange", "blue", "green", "red", "grey", "purple"];
for (const color of colors) {
    entry_points[`tlp-${color}`] = `./src/scss/tlp-${color}.scss`;
    entry_points[`tlp-${color}-condensed`] = `./src/scss/tlp-${color}-condensed.scss`;
}

const tlp_framework_config = {
    entry: entry_points,
    context: path.resolve(__dirname),
    output: {
        path: path.resolve(__dirname, "dist/"),
        filename: "tlp-[chunkhash].[name].js",
        library: "tlp"
    },
    resolve: {
        alias: {
            select2: "select2/dist/js/select2.full.js"
        }
    },
    module: {
        rules: [
            webpack_configurator.configureBabelRule(webpack_configurator.babel_options_ie11),
            webpack_configurator.rule_scss_loader,
            webpack_configurator.rule_css_assets
        ]
    },
    plugins: [
        webpack_configurator.getCleanWebpackPlugin(),
        webpack_configurator.getManifestPlugin(),
        ...webpack_configurator.getCSSExtractionPlugins()
    ]
};

const tlp_doc_config = {
    entry: {
        style: "./doc/css/main.scss",
        script: "./doc/js/index.js"
    },
    context: path.resolve(__dirname),
    output: webpack_configurator.configureOutput(path.resolve(__dirname, "doc/dist/")),
    externals: {
        tlp: "tlp"
    },
    module: {
        rules: [
            webpack_configurator.configureBabelRule(webpack_configurator.babel_options_ie11),
            webpack_configurator.rule_scss_loader
        ]
    },
    plugins: [
        webpack_configurator.getCleanWebpackPlugin(),
        webpack_configurator.getManifestPlugin(),
        ...webpack_configurator.getCSSExtractionPlugins()
    ]
};

module.exports = [tlp_framework_config, tlp_doc_config];
