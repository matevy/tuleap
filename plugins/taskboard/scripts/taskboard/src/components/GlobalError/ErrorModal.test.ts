/*
 * Copyright (c) Enalean, 2019-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { Vue } from "vue/types/vue";
import { shallowMount, Wrapper } from "@vue/test-utils";
import { createTaskboardLocalVue } from "../../helpers/local-vue-for-test";
import { createStoreMock } from "../../../../../../../src/www/scripts/vue-components/store-wrapper-jest";
import ErrorModal from "./ErrorModal.vue";
import * as tlp from "tlp";
import { Modal } from "tlp";

jest.mock("tlp", () => {
    return {
        __esModule: true,
        modal: jest.fn()
    };
});

describe("ErrorModal", () => {
    let local_vue: typeof Vue;

    beforeEach(async () => {
        local_vue = await createTaskboardLocalVue();
    });

    function createWrapper(error_message: string): Wrapper<ErrorModal> {
        return shallowMount(ErrorModal, {
            localVue: local_vue,
            mocks: {
                $store: createStoreMock({
                    state: { error: { modal_error_message: error_message } }
                })
            }
        });
    }

    it("it warns user that something is wrong with a button to show details", () => {
        const actual_tlp = jest.requireActual("tlp");
        jest.spyOn(tlp, "modal").mockImplementation(actual_tlp.modal);
        const wrapper = createWrapper("Full error message with details");
        expect(wrapper.element).toMatchSnapshot();
    });

    it(`shows the modal when mounted`, () => {
        const modal_show = jest.fn();
        jest.spyOn(tlp, "modal").mockImplementation(() => {
            return ({
                show: modal_show
            } as unknown) as Modal;
        });
        createWrapper("Full error message with details");
        expect(modal_show).toHaveBeenCalledTimes(1);
    });

    it("it display more details when user click on show error", () => {
        const error_message = "Full error message with details";
        const wrapper = createWrapper(error_message);

        wrapper.find("[data-test=show-details]").trigger("click");

        const details = wrapper.find("[data-test=details]");
        expect(details.text()).toEqual(error_message);
    });

    it("it warns user that something is wrong without any details", () => {
        const wrapper = createWrapper("");
        expect(wrapper.find("[data-test=show-details]").exists()).toBe(false);
        expect(wrapper.find("[data-test=details]").exists()).toBe(false);
    });
});
