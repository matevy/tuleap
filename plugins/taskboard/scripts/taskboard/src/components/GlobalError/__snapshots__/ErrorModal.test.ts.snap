// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal warns user that something is wrong with a button to show details 1`] = `
<div
  aria-labelledby="taskboard-error-modal-title"
  class="tlp-modal tlp-modal-danger tlp-modal-display tlp-modal-shown"
  role="dialog"
>
  <div
    class="tlp-modal-header"
  >
    <h1
      class="tlp-modal-title"
      id="taskboard-error-modal-title"
    >
      <i
        class="fas fa-exclamation-triangle tlp-modal-title-icon"
      />
       
      <translate-stub
        tag="span"
      >
        Oops, there's an issue
      </translate-stub>
    </h1>
     
    <div
      aria-label="Close"
      class="tlp-modal-close"
      data-dismiss="modal"
      role="button"
      tabindex="0"
    >
      
            Ã—
        
    </div>
  </div>
   
  <div
    class="tlp-modal-body"
  >
    <p
      data-current-language="en_US"
      data-msgid="It seems an action you tried to perform can't be done"
    >
      It seems an action you tried to perform can't be done
    </p>
     
    <a
      class="taskboard-error-modal-link"
      data-current-language="en_US"
      data-msgid="
                Show error details
            "
      data-test="show-details"
    >
      
                Show error details
            
    </a>
     
    <!---->
  </div>
   
  <div
    class="tlp-modal-footer"
  >
    <button
      class="tlp-button-danger tlp-button-outline tlp-modal-action"
      data-current-language="en_US"
      data-dismiss="modal"
      data-msgid="
            Close
        "
      type="button"
    >
      
            Close
        
    </button>
     
    <button
      class="tlp-button-danger tlp-modal-action"
      type="button"
    >
      <i
        class="fas fa-sync tlp-button-icon"
      />
       
      <translate-stub
        tag="span"
      >
        Reload the page
      </translate-stub>
    </button>
  </div>
</div>
`;
